File: src_v3/danbooru/gelbooru_client.py
import random

from danbooru.danbooru_client import DanbooruClient


class GelbooruClient(DanbooruClient):
    def _get_index_params(self):
        page = random.randint(1, 200)
        return {"page": "dapi", "s": "post", "q": "index", "json": "1", "limit": "100", "pid": str(page)}

    _title = "Gelbooru"
    _index_url = "https://gelbooru.com/index.php"

    def _get_full_post_url(self, post):
        return "https://gelbooru.com/index.php?page=post&s=view&id=" + str(post["id"])

    def _get_tag_list(self, post):
        return post["tags"].split()

File: src/main/kotlin/com/github/hannesbraun/katarina/modules/danbooru/GelbooruClient.kt
package com.github.hannesbraun.katarina.modules.danbooru

import com.google.gson.JsonObject
import kotlinx.coroutines.CoroutineScope
import kotlin.random.Random

class GelbooruClient(scope: CoroutineScope) : DanbooruClient(scope) {
    override val indexUrl = "https://gelbooru.com/index.php"
    override val name = "Gelbooru"

    override fun getIndexParameters(): List<Pair<String, String>> {
        val pid = Random.nextInt(1, 201)
        return listOf(
            Pair("page", "dapi"),
            Pair("s", "post"),
            Pair("q", "index"),
            Pair("json", "1"),
            Pair("limit", postBufferSize.toString()),
            Pair("pid", pid.toString())
        )
    }

    override fun generatePost(
        id: String, fileUrl: String, rating: String, score: String, tagList: List<String>
    ) = GelbooruPost(id, fileUrl, rating, score, tagList)

    override fun getTagListFromJson(post: JsonObject): List<String> = post.getString("tags").split(" ")
}

class GelbooruPost(id: String, fileUrl: String, rating: String, score: String, tagList: List<String>) : DanbooruPost(
    id,
    fileUrl,
    rating,
    score,
    tagList
) {
    override fun getFullPostUrl() = "https://gelbooru.com/index.php?page=post&s=view&id=$id"
}